pinot-minion/src/main/java/com/linkedin/pinot/minion/executor/BaseTaskExecutor.java

```diff
@@ -32,4 +40,22 @@ public void setMinionContext(@Nonnull MinionContext minionContext) {
   public void cancel() {
     _cancelled = true;
   }
+
+  @Override
+  public int uploadSegment(String uri, final String fileName, final InputStream inputStream, final long lengthInBytes,
+      FileUploadUtils.SendFileMethod httpMethod, String originalSegmentCrc, String jobType) {
+    List<Header> headers = new ArrayList<>();
+    Header ifMatchHeader = new Header(HttpHeaders.IF_MATCH, originalSegmentCrc);
+    HttpParams httpParams = DefaultHttpParams.getDefaultParams();
+    String userAgentParameter = String.valueOf(httpParams.getParameter("http.useragent"));
+
+    // TODO: Add proper version instead of hardcoding
+    userAgentParameter += " Pinot-Minion-" + jobType + "/1.0";
```

Comment:
please use static final strings for "Pinot-Minion" and the version.

We will fill in the version later.